src/shared/components/MSKTabs/MSKTabs.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import classnames from 'classnames';
+import ReactChild = React.ReactChild;
+import ReactNode = React.ReactNode;
+import ReactChildren = React.ReactChildren;
+
+interface IMSKTabProps {
+    hidden?:Boolean;
+    id:string;
+    linkText:string;
+    activeId?:string;
+}
+
+export class MSKTab extends React.Component<IMSKTabProps,{}> {
+
+    constructor(props: IMSKTabProps){
+        super(props);
+    }
+
+    render(){
+        const hidden = ((this.props.hidden===true) ? 'hidden' : '');
+        return <div className={classnames({ 'msk-tab':true, 'hidden':(this.props.hidden === true)  })}>{this.props.children}</div>
+    }
+
+}
+
+interface IMSKTabsState {
+    activeTabId:string;
+}
+
+interface IMSKTabsProps {
+    className?:string;
+    id?:string;
+    activeTabId?:string;
+    onTabClick(tabId:string):void;
+}
+
+export class MSKTabs extends React.Component<IMSKTabsProps, IMSKTabsState> {
+
+    private shownTabs:string[] = [];
+
+    constructor(){
+        super();
+        this.state = { activeTabId: "one" };
+    }
+
+    setActiveTab(id: string){
+        this.props.onTabClick(id);
+    }
+
+    render(){
+        if (this.props.children) {
+
+            let children: any[] = (this.props.children as any[]);
+
+            let hasActive: boolean = false;
+            let defaultTab: string;
+
+            const arr: any = _.reduce(React.Children.toArray(children), (memo: any, child: any) => {
```

Comment:
A component's props are immutable.  So this is the only way (in my research) for a parent to modify a child's props.  The clone is shallow so the DOM and children are merged not cloned.  I've verified that the tabs children are NOT re-instantiated.  Perhaps worth more research, but I think this is actually equivaent to what happens when declaratively in render. (not sure about that)